### PR TITLE
Read Iptables-save output in a more-memory-efficient way

### PR DESCRIPTION
**Pull request by ArtfulCoder**

### CODE REVIEW

File: pkg/proxy/iptables/proxier.go [hunk 1]
```diff
@@ -765,33 +765,84 @@ func makeChainLine(chain utiliptables.Chain) string {
...
+
+	for ; currentReadIndex < len(byteArray); currentReadIndex++ {
+		if byteArray[currentReadIndex] == ' ' {
+			// set rightTrimIndex
+			if rightTrimIndex == -1 {
+				rightTrimIndex = currentReadIndex
+			}
+		} else if (byteArray[currentReadIndex] == '\n') || (currentReadIndex == (len(byteArray) - 1)) {
+			// end of line or byte buffer is reached
+			if currentReadIndex <= leftTrimIndex {
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go]: how could `currentReadIndex` ever be less than `leftTrimIndex` ?

Comment (ArtfulCoder) [inline pkg/proxy/iptables/proxier.go]: It can't be less.
I still need to handle to == case.

**Comment by thockin** [comment]
LGTM

Unit test?

**Comment by ArtfulCoder** [comment]
am almost done with the unit test
will update PR

**Comment by ArtfulCoder** [comment]
Unit test added

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**